datadog/resource_datadog_dashboard.go

```diff
@@ -305,13 +305,13 @@ func buildTerraformTemplateVariables(datadogTemplateVariables *[]datadogV1.Dashb
 	terraformTemplateVariables := make([]map[string]string, len(*datadogTemplateVariables))
 	for i, templateVariable := range *datadogTemplateVariables {
 		terraformTemplateVariable := map[string]string{}
-		if v, ok := templateVariable.GetNameOk(); ok {
+		if v, ok := templateVariable.GetNameOk(); ok && v != nil {
 			terraformTemplateVariable["name"] = *v
 		}
-		if v, ok := templateVariable.GetPrefixOk(); ok {
+		if v, ok := templateVariable.GetPrefixOk(); ok && v != nil {
```

Comment:
```suggestion
		if v := templateVariable.GetPrefix(); len(v) > 0 {
```

I think that makes more sense in this case, the Ok doesn't really help us. You need to change the next line assignment though.